### PR TITLE
add imds asg target lifecycle hook

### PR DESCRIPTION
**Pull request by pete911**
**Issue #, if available:**
ASG target lifecycle hook is already available in [instance metadata service](https://docs.aws.amazon.com/autoscaling/ec2/userguide/retrieving-target-lifecycle-state-through-imds.html). This pull request adds it to IMDS option.
**Description of changes:**
Adding ASG Termination Lifecycle Hook to IMDS Processor.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by pete911** [comment]
I have create issue as well to discuss/describe this PR - https://github.com/aws/aws-node-termination-handler/issues/837

**Comment by pete911** [comment]
any update on this? If it is not required, I can close the PR and issue.

**Comment by ChristopherEdwardsNTC** [comment]
This would be super helpful!

File: pkg/monitor/asglifecycle/asg-lifecycle-monitor.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		return nil, fmt.Errorf("There was a problem creating an event ID from the event: %w", err)
+	}
+
+	return &monitor.InterruptionEvent{
+		EventID:      fmt.Sprintf("spot-itn-%x", hash.Sum(nil)),
+		Kind:         monitor.ASGLifecycleKind,
+		Monitor:      ASGLifecycleMonitorKind,
+		StartTime:    interruptionTime,
+		NodeName:     nodeName,
+		Description:  "AST tareget lifecycle state received. Instance will be \n",
```
Comment (Lu-David) [inline pkg/monitor/asglifecycle/asg-lifecycle-monitor.go:90]: nit: typo

Comment (Lu-David) [inline pkg/monitor/asglifecycle/asg-lifecycle-monitor.go:90]: nit: typo and incomplete description

**Comment by LikithaVemulapalli** [comment]
closing the PR as this change was included as part of another PR with e2e tests including your changes, once again thanks for working on the implementation :) 
https://github.com/aws/aws-node-termination-handler/pull/1067

**Pull request closed without merging**